llama_index/llms/ollama.py

```diff
@@ -0,0 +1,150 @@
+from typing import Any, Callable, Dict, Optional, Sequence, List, Iterator
+import json
+from llama_index.bridge.pydantic import Field, PrivateAttr
+
+from llama_index.callbacks import CallbackManager
+from llama_index.constants import DEFAULT_CONTEXT_WINDOW, DEFAULT_NUM_OUTPUTS
+from llama_index.llms.base import (
+    ChatMessage,
+    ChatResponse,
+    ChatResponseGen,
+    CompletionResponse,
+    CompletionResponseGen,
+    LLMMetadata,
+    llm_chat_callback,
+    llm_completion_callback,
+)
+from llama_index.llms.custom import CustomLLM
+from llama_index.llms.generic_utils import completion_response_to_chat_response
+from llama_index.llms.generic_utils import (
+    messages_to_prompt as generic_messages_to_prompt,
+)
+from llama_index.llms.generic_utils import stream_completion_response_to_chat_response
+
+
+class Ollama(CustomLLM):
+    base_url: str = "http://localhost:11434"
+    """Base url the model is hosted under."""
+
+    model: str = Field(description="The Ollama model to use.")
+    temperature: float = Field(description="The temperature to use for sampling.")
+    context_window: int = Field(
+        description="The maximum number of context tokens for the model."
+    )
+    prompt_key: str = Field(description="The key to use for the prompt in API calls.")
+    additional_kwargs: Dict[str, Any] = Field(
+        default_factory=dict, description="Additonal kwargs for the Ollama API."
+    )
+
+    _messages_to_prompt: Callable = PrivateAttr()
+    _completion_to_prompt: Callable = PrivateAttr()
+
+    def __init__(
+        self,
+        model: str,
+        temperature: float = 0.75,
+        additional_kwargs: Optional[Dict[str, Any]] = None,
+        context_window: int = DEFAULT_CONTEXT_WINDOW,
+        prompt_key: str = "prompt",
+        messages_to_prompt: Optional[Callable] = None,
+        completion_to_prompt: Optional[Callable] = None,
+        callback_manager: Optional[CallbackManager] = None,
+    ) -> None:
+        self._messages_to_prompt = messages_to_prompt or generic_messages_to_prompt
+        self._completion_to_prompt = completion_to_prompt or (lambda x: x)
+
+        super().__init__(
+            model=model,
+            temperature=temperature,
+            additional_kwargs=additional_kwargs or {},
+            context_window=context_window,
+            prompt_key=prompt_key,
+            callback_manager=callback_manager,
+        )
+
+    @classmethod
+    def class_name(cls) -> str:
+        """Get class name."""
+        return "Ollama_llm"
+
+    @property
+    def metadata(self) -> LLMMetadata:
+        """LLM metadata."""
+        return LLMMetadata(
+            context_window=self.context_window,
+            num_output=DEFAULT_NUM_OUTPUTS,
```

Comment:
Does ollama not let you configure the number of output tokens?

Reply:
Yea, I'm not able to find this piece of information on the ollama docs. 
If this parameter is for limiting the output token, it feels like it can be configurable from user side  because Ollama is simply taking the vanilla llama2 model from restAPI call at line 126

Reply:
Found the ollama doc here - they indeed don't have the said configuration: https://github.com/jmorganca/ollama/blob/main/docs/api.md

Reply:
Thanks for tracking that down!